Spec/RealtimeClient.swift

```diff
@@ -333,6 +333,53 @@ class RealtimeClient: QuickSpec {
                     expect(secs).to(beLessThanOrEqualTo(UInt(options.suspendedRetryTimeout)))
                 }
             }
+
+            it("should never register any connection listeners for internal use with the public EventEmitter") {
```

Comment:
Well it's a best effort to check the number of event listeners registered on the connection.  Do you have a better suggestion?  It's a very hard one to prove really, but at least this ensures common failures may be caught. WDYT?